driver/driver.feature

```diff
@@ -0,0 +1,1265 @@
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-write
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open read transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-read
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection has database: created-after-read
+
+
+  Scenario: Driver processes transaction commit errors correctly
+    Given connection open schema transaction for database: typedb
+    When typeql schema query
+      """
+      define attribute name;
+      """
+    Then transaction commits; fails
+
+
+  # TODO: check errors on transaction commits with callbacks (on_close) set!
+
+  ###########
+  # QUERIES #
+  ###########
+
+  Scenario: Driver processes ok query answers correctly
+    Given connection open schema transaction for database: typedb
+    When get answers of typeql schema query
+      """
+      define entity person;
+      """
+    Then answer type is: ok
+    Then answer type is not: concept rows
+    Then answer type is not: concept trees
+    Then result is a successful ok
+    Then transaction commits
+
+
+    # TODO: Test optionals when introduced
+  Scenario: Driver processes concept row query answers correctly
+    Given connection open schema transaction for database: typedb
+    Given typeql schema query
+      """
+      define entity person owns name @key; attribute name, value string;
+      """
+    When get answers of typeql read query
+      """
+      match entity $p;
+      """
+    Then answer type is: concept rows
+    Then answer type is not: ok
+    Then answer type is not: concept trees
+    Then answer query type is read
+    Then answer query type is not schema
+    Then answer query type is not write
+    Then answer size: 1
+    Then answer column names are:
+      | p |
+    Then answer get row(0) get entity type(p) get label: person
+    Then answer get row(0) get entity type by index of variable(p) get label: person
```

Comment:
I don't want to rely on header's ordering, so a safe way (+ more scalable in the code!!!) is to just say "get it by index of this variable in the header".

Reply:
Fair enough, and your whole line here composes well. Let's perhaps use commas to make it a bit more readable?
`answer get row(0), get entity type by index of variable(p), get label: person`

Reply:
I didn't use commas in concept API bdds and tried to remove them as much as possible. The reason is that I believe that constructions like `get smth(key) ...` are already separated because of `(key)`, and it's much easier to read for me, the stages of unwrapping are clear, and there is no redundancy. 

Don't you think so?

Reply:
I dont want to make more work for you, so we can keep it this way. If we have a multipart step in the future that doesn't have `()`, then i'd probably introduce commas everywhere

Reply:
Okay